### PR TITLE
feat: configurable adapter includes

### PR DESCRIPTION
**Pull request by dunqan**
Closes [HRZ-90345](https://spryker.atlassian.net/browse/HRZ-90345)

[HRZ-90345]: https://spryker.atlassian.net/browse/HRZ-90345?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/platform/core/src/services/providers.ts [hunk 1]
```diff
@@ -108,5 +109,9 @@ export const coreProviders: Provider[] = [
     provide: AppInitializer,
     useExisting: PageMetaResolverService,
   },
+  {
+    provide: IncludesService,
```
Comment (tobi-or-not-tobi) [inline libs/platform/core/src/services/providers.ts]: What about `JsonIncludesService`? Or `AdapterIncludesService`? Or `HttpIncludesService`?

File: libs/domain/product/src/services/list/adapter/product-list.adapter.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ export interface ProductListAdapter {
 }
 
 export const ProductListAdapter = 'oryx.ProductListAdapter';
+export const ProductListResource = 'product-list';
```
Comment (tobi-or-not-tobi) [inline libs/domain/product/src/services/list/adapter/product-list.adapter.ts]: Why not `oryx.ProductListResource`?

File: libs/domain/product/src/models/product-qualifier.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export interface ProductQualifier {
   sku?: string;
+  /** @deprecated since 1.3 */
```
Comment (tobi-or-not-tobi) [inline libs/domain/product/src/models/product-qualifier.ts]:
```suggestion
  /** @deprecated since 1.4 */
```

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/domain/product/src/services/list/adapter/product-list.adapter.ts (hunk 2 above)
Comment (dunqan) [reply libs/domain/product/src/services/list/adapter/product-list.adapter.ts]: 👍

File: libs/domain/product/src/models/product-qualifier.ts (hunk 3 above)
Comment (dunqan) [reply libs/domain/product/src/models/product-qualifier.ts]: 👍

File: libs/platform/core/src/services/providers.ts (hunk 1 above)
Comment (dunqan) [reply libs/platform/core/src/services/providers.ts]: JsonApiIncludesService?

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**